{Ng Yee Thern} iP

AugustDespair opened the pull request:
# DukeBot

> Have you ever wanted a bot to manage all your To-Dos? No? Yes. - Someone, probably.

DukeBot is a super **easy to use**, **text based CLI** chat bot that helps you manage your to-do lists. 

Features:
1. List all your to-dos
2. Add up to 3 types of tasks: to-dos, events and deadlines!
3. Mark your tasks as done
4. Deleting tasks

How to start *duking*:
- download from [this page](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html)
- open your terminal & `cd` to the right directory
- run ```java -jar ip.jar```

Future features:
- [X] Support finding a task based on name. 
- [ ] Support finding list of deadlines with same deadline. 

Check out this cool main method :nerd_face: :sunglasses: :
```java 

public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

AugustDespair committed:
Echoes Command

AugustDespair committed:
Added Add and List functionality

AugustDespair committed:
Added ability to mark as done/undone

AugustDespair committed:
Added Todo, Deadlines and Events

AugustDespair committed:
Edited the .sh file to support text UI testing

AugustDespair committed:
Added simple error handling

AugustDespair committed:
Added an exception class

AugustDespair committed:
Fixed bugs with DukeException

AugustDespair committed:
Added delete function

AugustDespair committed:
Add OOP

AugustDespair committed:
Add Save and Load Functionality

AugustDespair committed:
Add LocalDateTime support

AugustDespair committed:
Add Saving and Date features

AugustDespair committed:
Add some changes

AugustDespair committed:
Add OOP and Reorganise Code

AugustDespair committed:
Organise into packages

AugustDespair committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

AugustDespair committed:
Add Gradle

AugustDespair committed:
Add basic JUnit Tests

AugustDespair committed:
Add JavaDoc to files

AugustDespair committed:
Merge branch 'branch-A-JavaDoc'

AugustDespair committed:
Add GUI

AugustDespair committed:
Merge branch 'branch-Level-10'

AugustDespair committed:
Add Checkstyle

AugustDespair committed:
Add GUI and Checkstyle

joosxi commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,52 @@
+public class Main extends Application {
+
+    /**
+     * Start the GUI.
+     * @param stage
```
According to the coding standards there should be an empty line between the description and parameters section!

joosxi replied:
Perhaps a description could have been given for the parameter 'stage' for others to understand this method more easily!

joosxi commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,25 @@
+public abstract class Command {
+
+    /**
+     * Executes the command.
+     * @param tasks Current list of to-do-list tasks.
```
Same as above, should have an empty line between the description section and parameters section!

joosxi replied:
I like that there was punctuation and naming of the parameters are apropriate, eg 'tasks' to represent the list of tasks

joosxi commented on src/main/java/duke/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,53 @@
+public class DeadlineCommand extends Command {
+
+    /**
+     * Executes the deadline-creating command.
+     * @param taskList
```
Should have empty line between description section and parameters section!

joosxi commented on src/main/java/duke/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,53 @@
+        Deadline deadlineTask = new Deadline(deadline.toString(), date);
+        taskList.add(deadlineTask);
+
+//        System.out.printf("\t %s\n", deadlineTask);
```
Perhaps this comment could have been removed since it seems to be unused code!

joosxi commented on src/main/java/duke/commands/HelpCommand.java:
```diff
@@ -0,0 +1,46 @@
+        StringBuilder ret = new StringBuilder();
+
+        String[] commands = new String[]{"help: prints the help menu",
+            "list: prints out the todo list",
+            "bye: closes the program",
+            "mark: marks task as done",
+            "unmark: marks task as not done",
+            "todo: adds a todo task",
+            "find: finds all tasks containing given keyword",
+            "deadline: adds a task with a deadline",
+            "event: adds an event"
```
Indentation for these lines should be 8 spaces since they are wrapped lines!

joosxi reviewed (commented):
Overall the code is very nicely formatted, with empty lines breaking up long blocks of code. Just a few minor coding standards violations! Great job!!

xiaobill8 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,61 @@
+        case "find":
+            return new FindCommand();
+        case "todo":
+            if (commandInput.length < 2) {
```
Perhaps the handling of this edge case should be within TodoCommand, i.e. throwing and catching an exception

xiaobill8 commented on src/main/java/duke/data/Duke.java:
```diff
@@ -0,0 +1,71 @@
+                terminate = c.isEnd();
+            } catch (DukeException de) {
+                System.out.println(de.getMessage());
+            } catch (IndexOutOfBoundsException ioobe) {
```
Perhaps the exception handilng could be done at the Command level, as each command has a different command format to follow

xiaobill8 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,79 @@
+    public Deadline(String name, String dateAndTime) {
+        super(name);
+
+        String[] dateAndTimeSplit = dateAndTime.split(" ");
```
Perhaps using DateTimeFormatterBuilder to build a DateTimeFormatter that accepts custom formats, then using LocalDate.parse() with the formatter could be more effective

xiaobill8 reviewed (commented):
Well-written code in general! Good use of OOP principles

xiaobill8 commented on src/main/java/duke/commands/MarkCommand.java:
```diff
@@ -0,0 +1,56 @@
+ */
+public class MarkCommand extends Command {
+
+    private boolean mark;
```
Perhaps isMarked instead to follow the naming convention of booleans

xiaobill8 commented on src/main/java/duke/commands/EventCommand.java:
```diff
@@ -0,0 +1,53 @@
+            if (splitInput[i].equals("/at")) {
+                break;
+            }
+//            event.append(" " + splitInput[i]);
```
Perhaps redundant comment here could be removed

xiaobill8 commented on src/main/java/duke/commands/MarkCommand.java:
```diff
@@ -0,0 +1,56 @@
+        if (this.mark) {
+            taskList.get(index).completed();
+            sb.append("Nice! I've marked this task as done:\n");
+
```
Unncessary empty line here

xiaobill8 commented on src/main/java/duke/commands/MarkCommand.java:
```diff
@@ -0,0 +1,56 @@
+        } else {
+            taskList.get(index).uncompleted();
+            sb.append("OK, I've marked this task as not done yet:\n");
+
```
Again, unnecessary empty line here!

xiaobill8 reviewed (commented):
Sorry, realised that PR review should be on coding standards, so I've added some comments regarding that

AugustDespair committed:
Add Streams

FindCommand used a for loop to search for tasks containing a particular keyword.

Let's use streams to simplify the code further.

AugustDespair committed:
Add Assertions

FindCommand and DeadlineCommand requires a user to input both a command and a task name.
Previously this was done inside Parser.
Let's use assert statements to enforce this requirement inside the classes instead.

AugustDespair committed:
Add Assert Statements

AugustDespair committed:
Merge pull request #2 from AugustDespair/branch-A-Assertions

Branch-A-Assertions

AugustDespair committed:
Merge pull request #3 from AugustDespair/branch-A-Streams

Add Streams

AugustDespair committed:
Check code quality

AugustDespair committed:
Merge branch 'master' into branch-A-CodeQuality

AugustDespair committed:
Merge pull request #4 from AugustDespair/branch-A-CodeQuality

Check code quality

AugustDespair committed:
Add Description to Help

AugustDespair committed:
Add Better GUI

AugustDespair committed:
Update README

AugustDespair committed:
Fix Bugs
